query-engine/connector-test-kit-rs/query-tests-setup/src/config.rs

```diff
@@ -215,7 +215,11 @@ impl TestConfig {
             #[cfg(unix)]
             {
                 use std::os::unix::fs::PermissionsExt;
-                if path.metadata().is_ok_and(|md| md.permissions().mode() & 0o111 == 0) {
+                let is_executable = match path.metadata() {
+                    Err(_) => false,
+                    Ok(md) => md.permissions().mode() & 0o111 != 0,
+                };
+                if !is_executable {
```

Comment:
Not necessarily a relevant change to this PR, but it's ok :)

Reply:
Yes, I discovered the previous construct was experimental :/ as much as I liked it better. Good catch!